src/main/scala/nl/knaw/dans/easy/pid/PidGeneratorApp.scala

```diff
@@ -15,20 +15,26 @@
  */
 package nl.knaw.dans.easy.pid
 
+import java.io.Closeable
+
 import scala.util.Try
 
 /**
  * Exposes the application's API to the application's driver (e.g. Command or ServiceStarter).
  *
  * @param wiring object that configures and wires together the application's components
  */
-class PidGeneratorApp(wiring: ApplicationWiring) {
+class PidGeneratorApp(wiring: ApplicationWiring) extends Closeable {
 
   def generate(pidType: PidType): Try[String] = {
     wiring.pidGenerator.generate(pidType)
   }
 
-  def destroy(): Try[Unit] = {
-    wiring.databaseAccess.closeConnectionPool()
+  def init(): Try[Unit] = {
+    wiring.databaseAccess.initConnectionPool()
+  }
+
+  override def close(): Unit = {
```

Comment:
`destroy` is renamed to `close` to comply with the interface. In order to propagate a possible exception, I added `unsafeGetOrThrow` (see below)